src/runtime_musl_workaround.jl

```diff
@@ -0,0 +1,148 @@
+struct musl_tls_module
+    next::Ptr{musl_tls_module}
+    image::Ptr{musl_tls_module}
+    len::Csize_t
+    size::Csize_t
+    align::Csize_t
+    offset::Csize_t
+end
+
+# This structure taken from `ldso/dynlink.c`
+# https://github.com/ifduyue/musl/blob/aad50fcd791e009961621ddfbe3d4c245fd689a3/ldso/dynlink.c#L53-L107
+struct musl_dso
+    # Things we find mildly interesting
+    base::Ptr{Cvoid}
+    name::Ptr{UInt8}
+
+    # The wasteland of things we don't care about
+    dynv::Ptr{Csize_t}
+    next::Ptr{musl_dso}
+    prev::Ptr{musl_dso}
+
+    phdr::Ptr{Cvoid}
+    phnum::Cint
+    phentsize::Csize_t
+
+    syms::Ptr{Cvoid}
+    hashtab::Ptr{Cvoid}
+    ghashtab::Ptr{Cvoid}
+    versym::Ptr{Int16}
+    strings::Ptr{UInt8}
+    syms_next::Ptr{musl_dso}
+    lazy_next::Ptr{musl_dso}
+    lazy::Ptr{Csize_t}
+    lazy_cnt::Csize_t
+
+    map::Ptr{Cuchar}
+    map_len::Csize_t
+
+    # We assume that dev_t and ino_t are always `uint64_t`, even on 32-bit systems.
+    dev::UInt64
+    ino::UInt64
+    relocated::Cchar
+    constructed::Cchar
+    kernel_mapped::Cchar
+    mark::Cchar
+    bfs_built::Cchar
+    runtime_loaded::Cchar
+    # NOTE: struct layout rules should insert two bytes of space here
+    deps::Ptr{Ptr{musl_dso}}
+    needed_by::Ptr{musl_dso}
+    ndeps_direct::Csize_t
+    next_dep::Csize_t
+    ctor_visitor::Cint
+    rpath_orig::Ptr{UInt8}
+    rpath::Ptr{UInt8}
+
+    tls::musl_tls_module
+    tls_id::Csize_t
+    relro_start::Csize_t
+    relro_end::Csize_t
+    new_dtv::Ptr{Ptr{Cuint}}
+    new_tls::Ptr{UInt8}
+    td_index::Ptr{Cvoid}
+    fini_next::Ptr{musl_dso}
+
+    # Finally!  The field we're interested in!
+    shortname::Ptr{UInt8}
+
+    # We'll put this stuff at the end because it might be interesting to someone somewhere
+    loadmap::Ptr{Cvoid}
+    funcdesc::Ptr{Cvoid}
+    got::Ptr{Csize_t}
+end
+
+function replace_musl_shortname(lib_handle::Ptr{Cvoid})
+    # First, find the absolute path of the library we're talking about
+    lib_path = abspath(dlpath(lib_handle))
+
+    # Load the DSO object, which conveniently is the handle that `dlopen()`
+    # itself passes back to us.  Check to make sure it's what we expect, by
+    # inspecting the `name` field.  If it's not, something has gone wrong,
+    # and we should stop before touching anything else.
+    dso = unsafe_load(Ptr{musl_dso}(lib_handle))
+    dso_name = abspath(unsafe_string(dso.name))
+    if dso_name != lib_path
+        @debug("Unable to synchronize to DSO structure", name=dso_name, path=lib_path)
+        return lib_handle
+    end
+
+    # If the shortname is not NULL, break out.
+    if dso.shortname != C_NULL
+        @debug("shortname != NULL!", ptr=shortname_ptr, value=unsafe_string(shortname_ptr))
+        return lib_handle
+    end
+
+    # Calculate the offset of `shortname` from the base pointer of the DSO object
+    shortname_offset = fieldoffset(musl_dso, findfirst(fieldnames(musl_dso) .== :shortname))
```

Comment:
This should be much faster:
```suggestion
    shortname_offset = fieldoffset(musl_dso, findfirst(==(:shortname), fieldnames(musl_dso)))
```
It stops as soon as it finds the first item and doesn't allocate the whole `BitVector`

Reply:
Lovely!  You learn something new each day.  :)